File: packages/shared/src/types/imaging.ts
/**
 * AI Medical Imaging Analysis types.
 * Chest X-ray TB screening, retinal scans, general radiology AI.
 */

/** Supported AI analysis types. */
export type ImagingAnalysisType = "chest_xray_tb" | "retinal_scan" | "chest_xray_general";

/** Finding severity levels. */
export type FindingSeverity = "normal" | "mild" | "moderate" | "severe" | "critical";

/** Single finding from AI image analysis. */
export interface ImagingFinding {
  label: string;
  description: string;
  confidence: number;
  severity: FindingSeverity;
  location: string | null;
  icd10_code: string | null;
}

/** AI imaging analysis request. */
export interface ImagingAnalysisRequest {
  imaging_order_id: string;
  analysis_type: ImagingAnalysisType;
  image_url: string;
  patient_age?: number | null;
  patient_sex?: string | null;
}

/** Complete AI imaging analysis result. */
export interface ImagingAnalysisResult {
  imaging_order_id: string;
  analysis_type: ImagingAnalysisType;
  findings: ImagingFinding[];
  overall_impression: string;
  recommendation: string;
  confidence_score: number;
  model_version: string;
  requires_clinician_review: boolean;
}

/** Analysis type metadata. */
export interface ImagingAnalysisTypeInfo {
  type: ImagingAnalysisType;
  name: string;
  description: string;
  modality: string;
}

File: services/api-gateway/app/routers/imaging.py
"""
AI Medical Imaging Analysis router.
Endpoints for AI-assisted chest X-ray TB screening, retinal scans, and general radiology.
Per CLAUDE.md: AI NEVER auto-commits to patient records — clinician sign-off required.
"""

import uuid

from fastapi import APIRouter, Depends
from sqlalchemy.ext.asyncio import AsyncSession
from structlog import get_logger

from app.auth import CurrentUser, get_current_user
from app.auth.license_check import require_module
from app.database import get_db
from app.services.imaging.analyzer import ImagingAnalyzer
from app.services.imaging.models import (
    ImagingAnalysisRequest,
    ImagingAnalysisResult,
    ImagingAnalysisType,
)

logger = get_logger(__name__)

router = APIRouter(dependencies=[Depends(require_module("radiology"))])


@router.post("/analyze", response_model=ImagingAnalysisResult)
async def analyze_image(
    request: ImagingAnalysisRequest,
    db: AsyncSession = Depends(get_db),
    current_user: CurrentUser = Depends(get_current_user),
) -> ImagingAnalysisResult:
    """
    Submit a medical image for AI analysis.
    Returns structured findings that REQUIRE clinician review before recording.

    @param request: Analysis request with image URL and type
    @param db: Database session
    @param current_user: Authenticated user from JWT
    @returns AI analysis result (requires_clinician_review=True always)
    """
    analyzer = ImagingAnalyzer()
    result = await analyzer.analyze(request)

    logger.info(
        "imaging_analysis_complete",
        order_id=request.imaging_order_id,
        type=request.analysis_type,
        findings_count=len(result.findings),
        confidence=result.confidence_score,
        facility_id=str(current_user.facility_id),
    )

    return result


@router.get("/analysis-types")
async def list_analysis_types(
    current_user: CurrentUser = Depends(get_current_user),
) -> list[dict]:
    """
    List available AI imaging analysis types.

    @param current_user: Authenticated user from JWT
    @returns Available analysis types with descriptions
    """
    return [
        {
            "type": ImagingAnalysisType.CHEST_XRAY_TB,
            "name": "Chest X-ray TB Screening",
            "description": "AI-assisted tuberculosis screening from chest radiographs",
            "modality": "XR",
        },
        {
            "type": ImagingAnalysisType.RETINAL_SCAN,
            "name": "Retinal Scan Analysis",
            "description": "Diabetic retinopathy grading and glaucoma risk assessment",
            "modality": "fundoscopy",
        },
        {
            "type": ImagingAnalysisType.CHEST_XRAY_GENERAL,
            "name": "General Chest X-ray",
            "description": "AI-assisted general chest radiograph interpretation",
            "modality": "XR",
        },
    ]
